app/src/main/java/com/dreamisi/moviebase/movieDetails/MoviesDetailsFragment.kt

```diff
@@ -7,55 +7,49 @@ import android.widget.ImageView
 import android.widget.TextView
 import androidx.core.os.bundleOf
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.LinearLayoutManager.HORIZONTAL
 import androidx.recyclerview.widget.RecyclerView
 import com.bumptech.glide.Glide
 import com.dreamisi.moviebase.R
-import com.dreamisi.moviebase.data.MovieRepositoryProvider
 import com.dreamisi.moviebase.data.models.Movie
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.launch
 
 class MoviesDetailsFragment : Fragment(R.layout.fragment_movies_details) {
 
     private var recycler: RecyclerView? = null
-    private val scope = CoroutineScope(Dispatchers.Main)
+    private val viewModel: MovieDetailsViewModel by viewModels {
+        MovieDetailsViewModelFactory(
+            arguments?.getInt(MOVIE_ID) ?: error("Empty ID of Movie")
+        )
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        val movieID = arguments?.getInt(MOVIE_ID) ?: error("Wrong type of MOVIE_ID")
         recycler = view.findViewById(R.id.actor_cards)
         recycler?.adapter = ActorsListAdapter(requireContext())
         recycler?.layoutManager = LinearLayoutManager(requireContext(), HORIZONTAL, false)
-        scope.launch {
-            val application = requireActivity().application as MovieRepositoryProvider
-            val repository = application.provideMovieRepository()
-            val movie = repository.loadMovie(movieID)
-            if (movie != null) {
-                updateData(view, movie)
-            }
-
-        }
+        viewModel.movieDetailsLiveData.observe(this.viewLifecycleOwner, this::updateData)
```

Comment:
А тот изменить на обычную лямбду
```suggestion
        viewModel.movieDetailsLiveData.observe(this.viewLifecycleOwner, { updateData(it, view) } )
```